rbd/admin/task.go

```diff
@@ -0,0 +1,141 @@
+	Progress      float64  `json:"progress"`
+	RetryAttempts int      `json:"retry_attempts"`
+	RetryTime     string   `json:"retry_time"`
+	RetryMessage  string   `json:"retry_message"`
+}
+
+func parseTaskResponse(res commands.Response) (TaskResponse, error) {
+	var taskResponse TaskResponse
+	err := res.NoStatus().Unmarshal(&taskResponse).End()
+
+	return taskResponse, err
+}
+
+func parseTaskResponseList(res commands.Response) ([]TaskResponse, error) {
+	var taskResponseList []TaskResponse
+	err := res.NoStatus().Unmarshal(&taskResponseList).End()
+
+	return taskResponseList, err
+}
+
+// AddFlatten adds a task to flatten a cloned image asynchronously in the background
+// based on the supplied image spec.
+//
+// Similar To:
+//  rbd task add flatten <image_spec>
+func (ta *TaskAdmin) AddFlatten(img ImageSpec) (TaskResponse, error) {
+	m := map[string]string{
+		"prefix":     "rbd task add flatten",
+		"image_spec": img.spec,
+		"format":     "json",
+	}
+	return parseTaskResponse(commands.MarshalMgrCommand(ta.conn, m))
+}
+
+// AddRemove adds a task to remove an image asynchronously in the background based
+// on the supplied image spec.
+//
+// Similar To:
+//  rbd task add remove <image_spec>
+func (ta *TaskAdmin) AddRemove(img ImageSpec) (TaskResponse, error) {
+	m := map[string]string{
+		"prefix":     "rbd task add remove",
+		"image_spec": img.spec,
+		"format":     "json",
+	}
+
+	return parseTaskResponse(commands.MarshalMgrCommand(ta.conn, m))
+}
+
+// AddTrashRemove a task to remove an image from the trash asynchronously in the background
+// based on the supplied image id spec.
+//
+// Similar To:
+//  rbd task add trash remove <image_id_spec>
+func (ta *TaskAdmin) AddTrashRemove(img ImageSpec) (TaskResponse, error) {
+	m := map[string]string{
+		"prefix":        "rbd task add trash remove",
+		"image_id_spec": img.spec,
+		"format":        "json",
+	}
+
+	return parseTaskResponse(commands.MarshalMgrCommand(ta.conn, m))
+}
+
+// List pending or running asynchronous tasks.
+//
+// Similar To:
+//  rbd task list
+func (ta *TaskAdmin) List() ([]TaskResponse, error) {
+	m := map[string]string{
+		"prefix": "rbd task list",
+		"format": "json",
+	}
+
+	return parseTaskResponseList(commands.MarshalMgrCommand(ta.conn, m))
+}
+
+// GetTaskByID returns pending or running asynchronous task using id.
+//
+// Similar To:
+//  rbd task list <task_id>
+func (ta *TaskAdmin) GetTaskByID(taskID string) (TaskResponse, error) {
+	m := map[string]string{
+		"prefix":  "rbd task list",
+		"task_id": taskID,
+		"format":  "json",
+	}
+
+	return parseTaskResponse(commands.MarshalMgrCommand(ta.conn, m))
+}
+
+// Cancel a pending or running asynchronous task.
+//
+// Similar To:
+//  rbd task cancel <task_id>
+func (ta *TaskAdmin) Cancel(taskID string) (TaskResponse, error) {
+	m := map[string]string{
+		"prefix":  "rbd task cancel",
+		"task_id": taskID,
+		"format":  "json",
+	}
+
+	return parseTaskResponse(commands.MarshalMgrCommand(ta.conn, m))
```

Comment:
This and a bunch of other functions seems to have an extra blank line before return. Normally, we use a line break like this to indicate a "pause" in the logic of the function - a place for the reader to see the function is separated into sub-blocks. Since I don't see that happening here I think we should remove these empty lines.